src/render/shaders/Textures.hpp

```diff
@@ -167,12 +167,90 @@ void main() {
 
 inline const std::string FRAGBLUR1 = R"#(
 #version 100
-precision mediump float;
+precision            mediump float;
 varying mediump vec2 v_texcoord; // is in 0-1
-uniform sampler2D tex;
+uniform sampler2D    tex;
+
+uniform float        radius;
+uniform vec2         halfpixel;
+uniform int          passes;
+uniform int          boost_colors;
+uniform float        saturation_boost;
+uniform float        brightness_boost;
+//
+// see http://alienryderflex.com/hsp.html
+const float Pr = 0.299;
+const float Pg = 0.587;
+const float Pb = 0.114;
+
+// Huge shout-out to @fadaaszhi for the original formula
+// https://www.desmos.com/3d/a88652b9a4
+// Y is "v" ( brightness ). X is "s" ( saturation )
+// Determines if high brightness or high saturation is more important
+const float a = 0.93;
+const float b = 0.28;
+const float c = 0.66; //  Determines the smoothness of the transition of unboosted to boosted colors
+//
+
+// http://www.flong.com/archive/texts/code/shapers_circ/
+float doubleCircleSigmoid(float x, float a) {
+    float min_param_a = 0.0;
+    float max_param_a = 1.0;
+    a                 = max(min_param_a, min(max_param_a, a));
+
+    float y = .0;
+    if (x <= a) {
+        y = a - sqrt(a * a - x * x);
+    } else {
+        y = a + sqrt(pow(1. - a, 2.) - pow(x - 1., 2.));
+    }
+    return y;
+}
 
-uniform float radius;
-uniform vec2 halfpixel;
+// https://www.shadertoy.com/view/wt23Rt
+vec3 saturate(vec3 v) {
+    return clamp(v, 0., 1.);
```

Comment:
??

Reply:
That is now removed with the new HSL function in [2cb73d6](https://github.com/hyprwm/Hyprland/pull/3650/commits/2cb73d603c659dabd4195e28d50ef5235c01d18e)